src/classes/frDonorTest.cls

```diff
@@ -239,8 +239,127 @@ public class frDonorTest {
+        System.assertEquals(syncedContact.Email, existing.Email, 'The email should not have changed since the NO_OVERWRITE conflict resolution was used');
+    }
+    
+    static testMethod void syncEntity_conflictResolution_overwriteNonNull() {
+        createMapping('firstName', 'FirstName');
+        createMapping('lastName', 'LastName');
+        createMapping('email', 'email', frModel.MAPPING_OVERWRITE_NON_NULL);
+        createMapping('address1', 'MailingStreet');
+        createMapping('city', 'MailingCity');
+        createMapping('state', 'MailingState');
+        createMapping('postalCode', 'MailingPostalCode');
+        createMapping('country', 'MailingCountry', frModel.MAPPING_OVERWRITE_NON_NULL);
+        
+        String newEmail = 'example.email.overwrite@example.com';
+        
+        Contact existing = getTestContact(true);
+        Map<String, Object> request = getTestRequest();
+        request.put('id', existing.fr_Id__c);
+        request.put('email', newEmail); //should overwrite existing value
+        request.put('country', null);  //should not overwrite existing value
+
+        frTestUtil.createTestPost(request);
+        Test.startTest();
+        frWSDonorController.syncEntity();        
+        Test.stopTest();
+        frTestUtil.assertNoErrors();
+        
+        Contact syncedContact = [SELECT Id, fr_ID__c, Email, MailingCountry FROM Contact WHERE fr_Id__c = :existing.fr_Id__c];
+        System.assertEquals(existing.Id, syncedContact.Id, 'The existing contact should have been used');
+        System.assertEquals(syncedContact.Email, newEmail, 'The email should have changed since the OVERWRITE_NON_NULL conflict resolution was used and the incoming value was non null');
+        System.assertEquals(syncedContact.MailingCountry, existing.MailingCountry, 'The country should NOT have changed since the OVERWRITE_NON_NULL conflict resolution was used and the incoming value was null');
+    }
+    
+    static testMethod void syncEntity_conflictResolution_overwriteMoreRecent_isMoreRecent() {
+        createMapping('firstName', 'FirstName');
+        createMapping('lastName', 'LastName');
+        createMapping('email', 'email', frModel.MAPPING_OVERWRITE_RECENT);
+        createMapping('address1', 'MailingStreet');
+        createMapping('city', 'MailingCity');
+        createMapping('state', 'MailingState');
+        createMapping('postalCode', 'MailingPostalCode');
+        createMapping('country', 'MailingCountry');
+        
+        String newEmail = 'example.email.overwrite@example.com';
+        
+        Contact existing = getTestContact(true);
+        Map<String, Object> request = getTestRequest();
+        request.put('id', existing.fr_Id__c);
+        request.put('email', newEmail); //should overwrite existing value
+        request.put('updtime', DateTime.now().addDays(1).getTime()); //one day in the future, so it should always be more recent
```

Comment:
Ok this was the only change I didn't implement from the feedback.  It was going to require us to requery the record from the database because that's how LastModifiedDate gets populated, but we know it was last modified at the time the test is running (aka DateTime.now()) since it gets inserted a few lines above this call so I think we should be fine.  If you feel strongly, I can definitely implement the requery-after-insert

Reply:
Oh that's fine, I thought it was coming from the `getTestContact` method. Pretty sure you'll never hit it anyways since that's a static value as is